datafusion/tests/sql.rs

```diff
@@ -1896,6 +1896,28 @@ async fn left_join() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn left_join_unbalanced() -> Result<()> {
+    // the t1_id is larger than t2_id so the hash_build_probe_order optimizer should kick in
+    let mut ctx = create_join_context_unbalanced("t1_id", "t2_id")?;
+    let equivalent_sql = [
+        "SELECT t1_id, t1_name, t2_name FROM t1 LEFT JOIN t2 ON t1_id = t2_id ORDER BY t1_id",
+        "SELECT t1_id, t1_name, t2_name FROM t1 LEFT JOIN t2 ON t2_id = t1_id ORDER BY t1_id",
```

Comment:
ANother good test would be to select columns in a different order than t1 and then t2 -- for example:
```
        "SELECT t2_name, t1_name, t1_id  FROM t1 LEFT JOIN t2 ON t2_id = t1_id ORDER BY t1_id",
```